lib/parameter_parser/email_alert_parameter_parser.rb

```diff
@@ -0,0 +1,98 @@
+# Ensures that email alert filter keys and values provided by users
+# and that Finder Frontend sends to Email Alert API are within the
+# set of permitted keys and values provided in the content item or
+# registries of people, organisations, taxons, and world locations.
+module ParameterParser
+  class EmailAlertParameterParser
+    include ActiveModel::Validations
+
+    validates :applied_filters, email_alert_params: true
+
+    def initialize(content_item, filter_params, params)
+      @content_item = content_item
+      @filter_params = parsed_params(filter_params, params).deep_stringify_keys
+    end
+
+    def applied_filters
+      @applied_filters ||= begin
+        permitted_filter_keys.each_with_object({}) { |key, filter_hash|
+          filter_hash[key] = permitted_values_for_filter_key(key)
+        }.compact
+      end
+    end
+
+  private
+
+    attr_reader :filter_params, :content_item
+
+    def permitted_facets
+      content_item.dig("details", "email_filter_facets") || []
+    end
+
+    def permitted_filter_keys
+      allowed = permitted_facets.map { |facet| key_for_facet(facet) }
+      given = filter_params.keys
+      allowed & given
+    end
+
+    def permitted_values_for_filter_key(key)
+      facet = find_facet_by_key(key)
+      return if facet.blank?
```

Comment:
`facet.blank?` can't ever be true here, right?  because `permitted_values_for_filter_key` is only called with values in `permitted_facets.keys`